From f35275add5b1df17b0708672f25222c23d99cacc Mon Sep 17 00:00:00 2001
From: Markus Kappeler <markus.kappeler@bytesatwork.ch>
Date: Tue, 26 Mar 2013 20:57:11 +0100
Subject: [PATCH] bytePANEL: enable NAND and EEPROM

Enabling NAND with 4bit ECC and using standard MTD partitions.
Additionally enable internal EEPROM on modul.
---
 arch/arm/mach-omap2/board-am335x-bytepanel.c |   56 ++++++++++++--------------
 1 file changed, 26 insertions(+), 30 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335x-bytepanel.c b/arch/arm/mach-omap2/board-am335x-bytepanel.c
index df0d46c..903f7e5 100644
--- a/arch/arm/mach-omap2/board-am335x-bytepanel.c
+++ b/arch/arm/mach-omap2/board-am335x-bytepanel.c
@@ -889,43 +889,35 @@ static void haptics_init(int evm_id, int profile)
 static struct mtd_partition am335x_nand_partitions[] = {
 /* All the partition sizes are listed in terms of NAND block size */
 	{
-		.name           = "SPL",
-		.offset         = 0,			/* Offset = 0x0 */
-		.size           = SZ_128K,
+		.name           = "xloader-nand",
+		.offset         = 0,
+		.size           = 4*(SZ_128K),
+		.mask_flags     = MTD_WRITEABLE
 	},
 	{
-		.name           = "SPL.backup1",
-		.offset         = MTDPART_OFS_APPEND,	/* Offset = 0x20000 */
-		.size           = SZ_128K,
+		.name           = "uboot-nand",
+		.offset         = MTDPART_OFS_APPEND,
+		.size           = 14*(SZ_128K),
+		.mask_flags     = MTD_WRITEABLE
 	},
 	{
-		.name           = "SPL.backup2",
-		.offset         = MTDPART_OFS_APPEND,	/* Offset = 0x40000 */
-		.size           = SZ_128K,
+		.name           = "params-nand",
+		.offset         = MTDPART_OFS_APPEND,
+		.size           = 2*(SZ_128K)
 	},
 	{
-		.name           = "SPL.backup3",
-		.offset         = MTDPART_OFS_APPEND,	/* Offset = 0x60000 */
-		.size           = SZ_128K,
+		.name           = "linux-nand",
+		.offset         = MTDPART_OFS_APPEND,
+		.size           = 40*(SZ_128K)
 	},
 	{
-		.name           = "U-Boot",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x80000 */
-		.size           = 15 * SZ_128K,
+		.name           = "jffs2-nand",
+		.offset         = MTDPART_OFS_APPEND,
+		.size           = 1040*(SZ_128K),	/*130MB*/
 	},
 	{
-		.name           = "U-Boot Env",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x260000 */
-		.size           = 1 * SZ_128K,
-	},
-	{
-		.name           = "Kernel",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x280000 */
-		.size           = 40 * SZ_128K,
-	},
-	{
-		.name           = "File System",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x780000 */
+		.name           = "usr-nand",
+		.offset         = MTDPART_OFS_APPEND,
 		.size           = MTDPART_SIZ_FULL,
 	},
 };
@@ -1016,7 +1008,7 @@ static void evm_nand_init(int evm_id, int profile)
 		&am335x_nand_timings);
 	if (!pdata)
 		return;
-	pdata->ecc_opt =OMAP_ECC_BCH8_CODE_HW;
+	pdata->ecc_opt = OMAP_ECC_BCH4_CODE_HW;
 	pdata->elm_used = true;
 	gpmc_device[0].pdata = pdata;
 	gpmc_device[0].flag = GPMC_DEVICE_NAND;
@@ -1204,6 +1196,7 @@ static struct evm_dev_cfg bytepanel_dev_cfg[] = {
 	{mmc0_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
 	{lcdc_init,     DEV_ON_BASEBOARD, PROFILE_NONE},
 	{spi0_init,     DEV_ON_BASEBOARD, PROFILE_NONE},
+        {evm_nand_init, DEV_ON_BASEBOARD, PROFILE_NONE},
 	{NULL, 0, 0},
 };
 
@@ -1281,10 +1274,13 @@ static struct tps65910_board am335x_tps65910_info = {
 };
 
 static struct i2c_board_info __initdata am335x_i2c0_boardinfo[] = {
-	{
+               {
+                I2C_BOARD_INFO("24c128", 0x50),
+               },
+               {
 		I2C_BOARD_INFO("tps65910", TPS65910_I2C_ID1),
 		.platform_data  = &am335x_tps65910_info,
-	},
+               },
 };
 
 static struct omap_musb_board_data musb_board_data = {
-- 
1.7.10.4

